stac_fastapi/api/setup.py

```diff
@@ -7,8 +7,8 @@
 
 install_requires = [
     "attrs",
-    "pydantic[dotenv]<2",
-    "stac_pydantic==2.0.*",
+    "pydantic[dotenv]>=2",
+    "stac_pydantic>=3",
```

Comment:
we should maybe restrict stac_pydantic to `<4.0`

Reply:
maybe we should just remove the requirements here and rely on `stac-fastapi.types` requirements?

maybe we should also pin `stac-fastapi.types` version

Reply:
Following [Henry Schreiner advice](https://iscinumpy.dev/post/bound-version-constraints/) I usually only add lower bounds.